homeassistant/helpers/translation.py

```diff
@@ -321,3 +330,47 @@ async def async_get_translations(
     for entry in cached:
         result.update(entry)
     return result
+
+
+@bind_hass
+def get_cached_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> dict[str, Any]:
+    """Return cached all backend translations.
+
+    If integrations specified, return translations for them.
+    Otherwise default to loaded integrations.
+    """
+    if integrations is not None:
+        components = set(integrations)
+    elif category in ("state", "entity_component", "services"):
+        components = set(hass.config.components)
+    else:
+        # Only 'state' supports merging, so remove platforms from selection
+        components = {
+            component for component in hass.config.components if "." not in component
+        }
+
+    result = {}
+
+    if TRANSLATION_FLATTEN_CACHE in hass.data:
+        cache = hass.data[TRANSLATION_FLATTEN_CACHE]
+        cached = cache.get_cached(language, category, components)
+        for entry in cached:
+            result.update(entry)
+
+    return result
+
+
+@bind_hass
+async def load_state_translations_to_cache(
+    hass: HomeAssistant,
+    language: str,
+) -> None:
+    """Load state translations to cache."""
+    await async_get_translations(hass, language, "entity")
```

Comment:
It seems very inefficient to call `async_get_translations` just to populate the cache.

Could we maybe add a function `async_get_translations` which calls `cache.async_load` instead of `cache.async_fetch`?

`async_load_translations` should be refactored so the code which prepares the set of components and the code which creates the translation cache is not duplicated in the new function `async_load_translations`

Reply:
I have refactored this logic and extracted `async_load_translations`